cardano-api/src/Cardano/Api/TxBody.hs

```diff
@@ -1213,29 +1212,32 @@ data TxOutDatum ctx era where
      -- only be used in the context of the transaction body, and does not occur
      -- in the UTxO. The UTxO only contains the datum hash.
      --
-     TxOutDatum'      :: ScriptDataSupportedInEra era
+     TxOutDatumInTx'  :: ScriptDataSupportedInEra era
                       -> Hash ScriptData
                       -> ScriptData
                       -> TxOutDatum CtxTx era
 
      -- | A transaction output that specifies the whole datum instead of the
-     -- datum hash.
-     TxOutInlineDatum :: InlineDatumSupportedInEra era
+     -- datum hash. Note that the datum map will not be updated with this datum,
+     -- it only exists at the transaction output.
+     --
+     TxOutDatumInline :: InlineDatumSupportedInEra era
                       -> ScriptData
                       -> TxOutDatum ctx era
 
 deriving instance Eq   (TxOutDatum ctx era)
 deriving instance Show (TxOutDatum ctx era)
 
-pattern TxOutDatum :: ScriptDataSupportedInEra era
-                   -> ScriptData
-                   -> TxOutDatum CtxTx era
-pattern TxOutDatum s d  <- TxOutDatum' s _ d
+pattern TxOutDatumInTx
+  :: ScriptDataSupportedInEra era
+  -> ScriptData
+  -> TxOutDatum CtxTx era
+pattern TxOutDatumInTx s d  <- TxOutDatumInTx' s _ d
```

Comment:
Since the line is changing we can take the opportunity to remove the extra space before `<-`.